Foundation/include/Poco/NotificationQueue.h

```diff
@@ -150,6 +150,7 @@ class Foundation_API NotificationQueue
 	NfQueue           _nfQueue;
 	WaitQueue         _waitQueue;
 	mutable FastMutex _mutex;
+	volatile bool     _cannotWait;
```

Comment:
what is the reason for `volatile` here?
also, seems like `_dontWait` would be a more appropriate name